lib/instance_billing_flavor_check/utils.py

```diff
@@ -240,25 +242,32 @@ def make_request(rmt_ip_addr, metadata, identifier):
             message = 'Request error:{}'.format(err)
         except Exception as err:
             message = 'Unexpected error: {}'.format(err)
-    
+
         if message:
-            logger.warning(
-                'Attempt {}: failed: {}'.format(retry_count, message)
-            )
-            if 'Timeout' in message:
-                retry_count += 1
+            if 'Timeout' in message or 'Connecting' in message:
+                logger.warning(
+                    'Attempt {}: failed: {}'.format(retry_count, message)
+                )
                 time.sleep(2)
-            if 'Timeout' in message or retry_count == 4:
-                return
-
-        if response.status_code == 200:
-            result = response.json()
-            logger.debug(result)
-            return result.get('flavor')
-
-    logger.error(
-        'Request to check if instance is PAYG/BYOS failed: %s', response.reason
-    )
+                retry_count += 1
+                continue
+            else:
+                # error is not time out => return None
+                logger.error(
```

Comment:
Switch to `warning` please. While there is an error with the request, we still create a deterministic result and we do not want to train people that look at logs to ignore error messages. The `warning` indicates that we are unsure about the true state of the flavor.